### ISSUE TITLE
[ISSUE] Issue with latest databricks provider and authentication

### ISSUE THREAD

**Issue by dzeediscoveryzone** [databricks/terraform-provider-databricks#1619]
### Configuration
```hcl
terraform {
  required_version = "~> 1.2.0"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.0"
    }
    null = {
      source  = "hashicorp/null"
      version = ">= 2"
    }
    random = {
      source  = "hashicorp/random"
      version = ">= 2"
    }
    archive = {
      source = "hashicorp/archive"
      version = ">= 2"
    }
    time = {
      source = "hashicorp/time"
      version = ">= 0.6"
    }
    databricks = {
      source  = "databricks/databricks"
      version = "~> 1.3"
    }
  }
}

```

### Expected Behavior

We're migrating from AWS Provider `3.74.3` and databrickslabs/databricks `0.5.1` to the latest for both however I'm receiving errors about authentication. I confirmed that the username/password work by logging directly into the portal. Databricks plan and apply should complete successfully but fails with errors.

### Actual Behavior

```hcl
╷
│ Error: cannot read instance pool: authentication is not configured for provider. Environment variables used: DATABRICKS_USERNAME, DATABRICKS_PASSWORD. Please check https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs#authentication for details
│
│   with module.workspace_resources.module.instance_pool["default_resource_pool"].databricks_instance_pool.this,
│   on .terraform/modules/workspace_resources/modules/databricks-instance-pool/main.tf line 2, in resource "databricks_instance_pool" "this":
│    2: resource "databricks_instance_pool" "this" {
│
╵
```

### Steps to Reproduce

Upgrade AWS and Databricks providers to the latest and run terraform plan. I confirmed that using the old versions works correctly.

### Terraform and provider versions

terraform version: `1.2.9`
databricks provider version:`1.3.0`

**Comment by alexott** [comment]
Please provide the content of the 

```hcl
provider "databricks" {
}
```

block

**Comment by dzeediscoveryzone** [comment]
Hi @alexott, please see below:

```hcl
provider "databricks" {
  host = "https://accounts.cloud.databricks.com"
}
```

**Comment by alexott** [comment]
how do you set username & password?

**Comment by dzeediscoveryzone** [comment]
via environment variables:

```shell
export DATABRICKS_USERNAME="<REDACTED>"
export DATABRICKS_PASSWORD="<REDACTED>"
```

**Comment by alexott** [comment]
Instance pool is the workspace level object, not account level object. You need to point your Databricks provider to a specific workspace

**Comment by dzeediscoveryzone** [comment]
> 

It this a new requirement for the latest provider version? The only difference in our deployment model is the version of the provider. I wasn't able to find how to set a workspace ID in the provider block.

**Comment by alexott** [comment]
you need to set host to the workspace itself

**Comment by dzeediscoveryzone** [comment]
> you need to set host to the workspace itself

Hi @alexott, that wasn't my issue specifically and I wanted to shed some light in case somebody else runs into this. I had to do a few things:

1. update all git terraform modules required versions
2. update internal modules inside of a git terraform module (I missed this initially)
3. removed .terraform.lock.hcl files from all environments (main deployment has 3 folder: dev,qa,prod)
4. go into each folder and run terraform/terragrunt state replace-provider databrickslabs/databricks databricks/databricks

Once this was updated I was able to plan and apply successfully.
